panda/src/navigation/navMeshBuilder.h

```diff
@@ -0,0 +1,164 @@
+#ifndef NAVMESHBUILDER_H
+#define NAVMESHBUILDER_H
+
+#include "Recast.h"
+#include <string>
+#include "geom.h"
+#include "geomNode.h"
+#include "nodePath.h"
+#include "pandaFramework.h"
+#include "pandaSystem.h"
+#include "navMesh.h"
+
+enum SamplePolyAreas {
+  SAMPLE_POLYAREA_GROUND,
+  SAMPLE_POLYAREA_WATER,
+  SAMPLE_POLYAREA_ROAD,
+  SAMPLE_POLYAREA_DOOR,
+  SAMPLE_POLYAREA_GRASS,
+  SAMPLE_POLYAREA_JUMP,
+};
+enum SamplePolyFlags {
+  SAMPLE_POLYFLAGS_WALK = 0x01,		// Ability to walk (ground, grass, road)
+  SAMPLE_POLYFLAGS_SWIM = 0x02,		// Ability to swim (water).
+  SAMPLE_POLYFLAGS_DOOR = 0x04,		// Ability to move through doors.
+  SAMPLE_POLYFLAGS_JUMP = 0x08,		// Ability to jump.
+  SAMPLE_POLYFLAGS_DISABLED = 0x10,		// Disabled polygon
+  SAMPLE_POLYFLAGS_ALL = 0xffff	// All abilities.
+};
+struct BuildSettings {
+  
+  float cell_size;
+  float cell_height;
+  float agent_height;
+  float agent_radius;
+  float agent_max_climb;
+  float agent_max_slope;
+  float region_min_size;
+  float region_merge_size;
+  float edge_max_len;
+  float edge_max_error;
+  float verts_per_poly;
+  float detail_sample_dist;
+  float detail_sample_max_error;
+  int partition_type;
+  float nav_mesh_bMin[3];
+  float nav_mesh_bMax[3];
+  float tile_size;
+};
+
+
+class NavMeshBuilder {
+private:
+  std::string _filename;
+  float _scale;
+  float *_verts;
+  int *_tris;
+  float *_normals;
+  int _vert_count;
+  int _tri_count;
+
+  void add_vertex(float x, float y, float z, int &cap);
+  void add_triangle(int a, int b, int c, int &cap);
+
+
+  void process_geom_node(GeomNode *geomnode, int &vcap, int &tcap);
+  void process_geom(CPT(Geom) geom, int &vcap, int &tcap);
+  void process_vertex_data(const GeomVertexData *vdata, int &vcap);
+  void process_primitive(const GeomPrimitive *orig_prim, const GeomVertexData *vdata, int &tcap);
+
+  std::map<LVector3, int> mp;
+  std::vector<LVector3> vc, fc;
+  bool loaded;
+  int index_temp;
```

Comment:
Members should be underscored.  Also, `mp` is not a very descriptive name; you should favour readability over terseness.

Reply:
Done